.github/workflows/gcr-cleaner.yaml

```diff
@@ -0,0 +1,32 @@
+# .github/workflows/gcr-cleaner.yaml
+name: 'gcr-cleaner'
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
+jobs:
+  gcr-cleaner:
+    runs-on: 'ubuntu-latest'
+    steps:
+      # configure based on registry
+      - uses: 'docker/login-action@v2'
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      # customized based on the gcr-cleaner flags
+      # -repo - list of repo to clean (required)
+      # -grace - relative duration in which to ignore references, refs newer than duration will not be deleted
+      # -keep - minimum number of images to keep, will not consider images inside the grace duration
+      # -dry_run - if set to true, will not delete anything and outputs what would have been deleted
+      # -recursive - if set to true, will recursively search all child repositories
```

Comment:
Excellent comment! So if I'm reading this correctly we'll always keep 30 days worth of images as well as the 10 most recent images that are older than 30 days?

Reply:
Correct. Here is the information from the [gcr-cleaner `README.md`](https://raw.githubusercontent.com/GoogleCloudPlatform/gcr-cleaner/main/README.md) as an FYI.

> - `grace` - Relative duration in which to ignore references. This value is
>   specified as a time duration value like "5s" or "3h". If set, refs newer than
>   the duration will not be deleted. If unspecified, the default is no grace
>   period (all untagged image refs are deleted).
> 
> - `keep` - If an integer is provided, it will always keep that minimum number of
>   images. Note that it will not consider images inside the `grace` duration. GCR
>   Cleaner attempts to keep the most recently created images, but there are some
>   caveats. Some community tooling sets container creation time to a date back in
>   1980, which breaks the default sorting algorithm. As such, GCR Cleaner uses
>   the following sorting algorithm for container images:
> 
>     - If either of the containers were created before Docker even existed, it
>       sorts by the date the container was uploaded to the registry.
> 
>     - If two containers were created at the same timestamp, it sorts by the date
>       the container was uploaded to the registry.
> 
>     - In all other situations, it sorts by the timestamp the container was
>       created.
> 
>   This algorithm exists to preserve ordering for containers that are moved
>   between registries.